WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Shipping Address Validation/ShippingLabelAddressFormViewController.swift

```diff
@@ -179,20 +228,26 @@ private extension ShippingLabelAddressFormViewController {
                                                                      placeholder: Localization.address2FieldPlaceholder,
                                                                      state: .normal,
                                                                      keyboardType: .default,
-                                                                     textFieldAlignment: .leading) { (newText) in
-
+                                                                     textFieldAlignment: .leading) { [weak self] (newText) in
+            self?.viewModel.handleAddressValueChanges(row: row, newValue: newText)
         }
         cell.configure(viewModel: cellViewModel)
     }
 
+    func configureFieldError(cell: BasicTableViewCell, row: Row) {
+        cell.textLabel?.text = viewModel.addressValidationError?.addressError
+        cell.textLabel?.textColor = .error
+        cell.backgroundColor = .listBackground
+    }
+
     func configureCity(cell: TitleAndTextFieldTableViewCell, row: Row) {
         let cellViewModel = TitleAndTextFieldTableViewCell.ViewModel(title: Localization.cityField,
                                                                      text: viewModel.address?.city,
                                                                      placeholder: Localization.cityFieldPlaceholder,
                                                                      state: .normal,
                                                                      keyboardType: .default,
-                                                                     textFieldAlignment: .leading) { (newText) in
-
+                                                                     textFieldAlignment: .leading) { [weak self] (newText) in
+            self?.viewModel.handleAddressValueChanges(row: row, newValue: newText)
```

Comment:
This call is repeated a few times. Would it make sense to extract a new method in the view controller that actually calls the viewModel? That way (and more about that in another comment) it would be easier to do things like adding a spinner when the call to the view model is executed